OutOfSchool/OutOfSchool.WebApi/Services/MinistryAdminService.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using Newtonsoft.Json;
+
+using OutOfSchool.Common;
+using OutOfSchool.Common.Models;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Config;
+using OutOfSchool.WebApi.Enums;
+using OutOfSchool.WebApi.Extensions;
+using OutOfSchool.WebApi.Models;
+using OutOfSchool.WebApi.Services.Communication;
+
+namespace OutOfSchool.WebApi.Services
+{
+    public class MinistryAdminService : CommunicationService, IMinistryAdminService
+    {
+        private readonly IdentityServerConfig identityServerConfig;
+        private readonly IEntityRepository<User> userRepository;
+        private readonly IMinistryAdminRepository ministryAdminRepository;
+        private readonly ILogger<MinistryAdminService> logger;
+        private readonly IMapper mapper;
+        private readonly ResponseDto responseDto;
+
+        public MinistryAdminService(
+            IHttpClientFactory httpClientFactory,
+            IOptions<IdentityServerConfig> identityServerConfig,
+            IOptions<CommunicationConfig> communicationConfig,
+            IMinistryAdminRepository ministryAdminRepository,
+            IEntityRepository<User> userRepository,
+            IMapper mapper,
+            ILogger<MinistryAdminService> logger)
+            : base(httpClientFactory, communicationConfig.Value)
+        {
+            this.identityServerConfig = identityServerConfig.Value;
+            this.ministryAdminRepository = ministryAdminRepository;
+            this.userRepository = userRepository;
+            this.mapper = mapper;
+            this.logger = logger;
+            responseDto = new ResponseDto();
+        }
+
+        public async Task<ResponseDto> CreateMinistryAdminAsync(string userId, CreateMinistryAdminDto ministryAdminDto, string token)
+        {
+            logger.LogDebug($"ministryAdmin creating was started. User(id): {userId}");
+
+            var request = new Request()
+            {
+                HttpMethodType = HttpMethodType.Post,
+                Url = new Uri(identityServerConfig.Authority, CommunicationConstants.CreateMinistryAdmin),
+                Token = token,
+                Data = ministryAdminDto,
+                RequestId = Guid.NewGuid(),
+            };
+
+            logger.LogDebug($"{request.HttpMethodType} Request(id): {request.RequestId} " +
+                $"was sent. User(id): {userId}. Url: {request.Url}");
+
+            var response = await SendRequest(request)
```

Comment:
Not a call but we already have a grpc communication, it has few benefits for future.